src/services/document-service.spec.js

```diff
@@ -114,4 +114,191 @@ describe('document-service', () => {
 
   });
 
+  describe('hardDeleteSection', () => {
+
+    describe('when a section has 3 revisions', () => {
+      let documentRevisionsBeforeDeletion;
+
+      beforeEach(async () => {
+        const unrelatedSection = {
+          key: uniqueId.create(),
+          type: 'markdown',
+          content: {
+            text: 'Unmodified text'
+          }
+        };
+
+        const sectionToBeDeleted = {
+          key: uniqueId.create(),
+          type: 'markdown',
+          content: {
+            text: 'Initial text'
+          }
+        };
+
+        documentRevisionsBeforeDeletion = await createTestRevisions(container, user, [
+          {
+            title: 'Revision 1',
+            slug: 'rev-1',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated A' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section A' } }
+            ]
+          },
+          {
+            title: 'Revision 2',
+            slug: 'rev-2',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated B' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section B' } }
+            ]
+          },
+          {
+            title: 'Revision 3',
+            slug: 'rev-3',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated C' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section B' } }
+            ]
+          },
+          {
+            title: 'Revision 4',
+            slug: 'rev-4',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated D' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section B' } }
+            ]
+          },
+          {
+            title: 'Revision 5',
+            slug: 'rev-5',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated E' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section C' } }
+            ]
+          }
+        ]);
+      });
+
+      describe('and only the second revision is hard-deleted', () => {
```

Comment:
A little confusing here with `the second revision` and `documentRevisionsBeforeDeletion[2]`

Reply:
True, maybe `the revision in the middle`?